### PR TITLE
yuzu/main: Specify string conversions explicitly

### PR DESCRIPTION
**Pull request by FearlessTobi**
Ports https://github.com/yuzu-emu/yuzu/pull/2513.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5010)

### CODE REVIEW

**Comment by wwylele** [comment]
ah, welp, I told myself if CI is fine then lgtm. Looks like it is not fine lol

File: src/citra_qt/util/spinbox.cpp [hunk 1]
```diff
@@ -185,9 +185,12 @@ void CSpinBox::UpdateText() {
 }
 
 QString CSpinBox::TextFromValue() {
-    return prefix + QString(HasSign() ? ((value < 0) ? "-" : "+") : "") +
-           QString("%1").arg(std::abs(value), num_digits, base, QLatin1Char('0')).toUpper() +
+    return prefix +
+           QString(HasSign() ? ((value < 0) ? QStringLiteral("-") : QStringLiteral("+"))
+                             : QStringLiteral("")) +
```
Comment (zhaowenlan1779) [inline src/citra_qt/util/spinbox.cpp]: 1. shouldn't be needing that `QString` constructor there
1. `QStringLiteral("")` should be `QString{}`.

File: src/citra_qt/util/spinbox.cpp [hunk 2]
```diff
@@ -185,9 +185,12 @@ void CSpinBox::UpdateText() {
...
+           QStringLiteral("%1").arg(std::abs(value), num_digits, base, QLatin1Char('0')).toUpper() +
            suffix;
+    return {};
```
Comment (zhaowenlan1779) [inline src/citra_qt/util/spinbox.cpp]: dead code

File: src/citra_qt/util/spinbox.cpp [hunk 3]
```diff
@@ -224,20 +227,20 @@ QValidator::State CSpinBox::validate(QString& input, int& pos) const {
...
 
     // Match digits corresponding to the chosen number base.
-    regexp += QString("[0-%1").arg(std::min(base, 9));
+    regexp.append(QStringLiteral("[0-%1").arg(std::min(base, 9)));
     if (base == 16) {
-        regexp += "a-fA-F";
+        regexp.append(QStringLiteral("a-fA-F"));
     }
-    regexp += "]";
+    regexp.append(QStringLiteral("]"));
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]:
```suggestion
    regexp.append(QLatin1Char(']'));
```

File: src/citra_qt/util/spinbox.cpp [hunk 4]
```diff
@@ -224,20 +227,20 @@ QValidator::State CSpinBox::validate(QString& input, int& pos) const {
...
 
     // Specify number of digits
     if (num_digits > 0) {
-        regexp += QString("{%1}").arg(num_digits);
+        regexp.append(QStringLiteral("{%1}").arg(num_digits));
     } else {
-        regexp += "+";
+        regexp.append(QStringLiteral("+"));
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]:
```suggestion
        regexp.append(QLatin1Char{'+'});
```

File: src/citra_qt/util/spinbox.cpp [hunk 5]
```diff
@@ -160,19 +160,19 @@ void CSpinBox::UpdateText() {
...
+        mask.append(QStringLiteral(">"));
 
         // Match num_digits digits
         // Digits irrelevant to the chosen number base are filtered in the validator
-        mask += QString("H").repeated(std::max(num_digits, 1));
+        mask.append(QStringLiteral("H").repeated(std::max(num_digits, 1)));
 
         // Switch off case conversion
-        mask += "!";
+        mask.append(QStringLiteral("!"));
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]:
```suggestion
        mask.append(QLatin1Char{'!'});
```

File: src/citra_qt/util/spinbox.cpp [hunk 6]
```diff
@@ -160,19 +160,19 @@ void CSpinBox::UpdateText() {
 
         // For base 10 and negative range, demand a single sign character
         if (HasSign())
-            mask += "X"; // identified as "-" or "+" in the validator
+            mask.append(QStringLiteral("X")); // identified as "-" or "+" in the validator
 
         // Uppercase digits greater than 9.
-        mask += ">";
+        mask.append(QStringLiteral(">"));
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]:
```suggestion
        mask.append(QLatin1Char{'>'});
```

File: src/citra_qt/util/spinbox.cpp [hunk 7]
```diff
@@ -160,19 +160,19 @@ void CSpinBox::UpdateText() {
 
         // For base 10 and negative range, demand a single sign character
         if (HasSign())
-            mask += "X"; // identified as "-" or "+" in the validator
+            mask.append(QStringLiteral("X")); // identified as "-" or "+" in the validator
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]:
```suggestion
            mask.append(QLatin1Char{'X'}); // identified as "-" or "+" in the validator
```

File: src/citra_qt/util/spinbox.cpp [hunk 8]
```diff
@@ -128,25 +128,25 @@ static QString StringToInputMask(const QString& input) {
...
-    mask.replace("#", "\\#");
-    mask.replace("H", "\\H");
-    mask.replace("h", "\\h");
-    mask.replace("B", "\\B");
-    mask.replace("b", "\\b");
-    mask.replace(">", "\\>");
-    mask.replace("<", "\\<");
-    mask.replace("!", "\\!");
+    mask.replace(QStringLiteral("\\"), QStringLiteral("\\\\"));
+    mask.replace(QStringLiteral("A"), QStringLiteral("\\A"));
```
Comment (lioncash) [inline src/citra_qt/util/spinbox.cpp]: A lot of these single character strings can be turned into QLatin1Char instances

File: src/citra_qt/main.cpp [hunk 9]
```diff
@@ -1781,19 +1796,19 @@ void GMainWindow::OnCoreError(Core::System::ResultStatus result, std::string det
         if (!details.empty()) {
             message = common_message.arg(QString::fromStdString(details));
         } else {
-            message = common_message.arg("A system archive");
+            message = common_message.arg(QStringLiteral("A system archive"));
```
Comment (lioncash) [inline src/citra_qt/main.cpp]: This should be a translatable string

File: src/citra_qt/main.cpp [hunk 10]
```diff
@@ -1781,19 +1796,19 @@ void GMainWindow::OnCoreError(Core::System::ResultStatus result, std::string det
...
         }
 
         title = tr("System Archive Not Found");
-        status_message = "System Archive Missing";
+        status_message = QStringLiteral("System Archive Missing");
```
Comment (lioncash) [inline src/citra_qt/main.cpp]: Ditto

File: src/citra_qt/main.cpp [hunk 11]
```diff
@@ -1781,19 +1796,19 @@ void GMainWindow::OnCoreError(Core::System::ResultStatus result, std::string det
...
     } else {
         title = tr("Fatal Error");
         message =
             tr("A fatal error occured. "
                "<a href='https://community.citra-emu.org/t/how-to-upload-the-log-file/296'>Check "
                "the log</a> for details."
                "<br/>Continuing emulation may result in crashes and bugs.");
-        status_message = "Fatal Error encountered";
+        status_message = QStringLiteral("Fatal Error encountered");
```
Comment (lioncash) [inline src/citra_qt/main.cpp]: Ditto

**Comment by FearlessTobi** [comment]
Rebased and Updated.

**Pull request merged**